src/expression.d

```diff
@@ -10852,7 +10852,29 @@ public:
         TypeVector tv = cast(TypeVector)tb;
         Type te = tv.elementType();
         dim = cast(int)(tv.size(loc) / te.size(loc));
-        return this;
+
+        bool checkElem(Expression elem)
+        {
+            if (elem.op == TOKint64 ||
+                elem.op == TOKfloat64 ||
+                elem.op == TOKcomplex80)
+                return false;
+
+             error("constant expression expected, not %s", elem.toChars());
+             return true;
+        }
+
+        e1 = e1.optimize(WANTvalue);
+        bool result;
+        if (e1.op == TOKarrayliteral)
+        {
+            foreach (i; 0 .. dim)
+                result |= checkElem((cast(ArrayLiteralExp)e1).getElement(i));
```

Comment:
Since this is an error checking loop, the idea is to check each of the AST nodes that are independent of each other. So the logic is currently correct.